qcodes/instrument/base.py

```diff
@@ -172,24 +172,7 @@ def add_parameter(self, name: str,
                 unit of the new parameter is inconsistent with the existing
                 one.
         """
-        existing_parameter = self.parameters.get(name, None)
-
-        if existing_parameter:
-
-            if not existing_parameter.abstract:
-                raise KeyError(f"Duplicate parameter name {name}")
-
-            existing_unit = getattr(existing_parameter, "unit", None)
-            new_unit = kwargs.get("unit", None)
-            if existing_unit != new_unit:
-                raise ValueError(
-                    f"The unit of the parameter '{name}' is '{new_unit}'. "
-                    f"This is inconsistent with the unit defined in the "
-                    f"base class"
-                )
-
-        param = parameter_class(name=name, instrument=self, **kwargs)
-        self.parameters[name] = param
+        parameter_class(name=name, instrument=self, **kwargs)
```

Comment:
We should probably be more defensive here and check that the parameter has been registred on the instruement. So that we better handle errors such as the ones fixed by #3187 

If not we could do one of.

* Register it here and warn that this is wrong e.g. the parameter subclass needs fixing
* Error out here and ask the user to fix the parameter class
* do 1 and turn it into 2 after some time

Reply:
> do 1 and turn it into 2 after some time
 :)

Reply:
Done